cupyx/scipy/linalg/decomp_lu.py

```diff
@@ -0,0 +1,174 @@
+import numpy
+
+import cupy
+from cupy import cuda
+from cupy.cuda import cublas
+from cupy.cuda import device
+from cupy.linalg import util
+
+if cuda.cusolver_enabled:
+    from cupy.cuda import cusolver
+
+
+def lu_factor(a, overwrite_a=False, check_finite=True):
+    """LU decomposition.
+
+    Decompose a given two-dimensional square matrix into ``P * L * U``,
+    where ``P`` is a permutation matrix,  ``L`` lower-triangular with
+    unit diagonal elements, and ``U`` upper-triangular matrix.
+    Note that in the current implementation ``a`` must be
+    a real matrix, and only float32 and float64 are supported.
+
+    Args:
+        a (cupy.ndarray): The input matrix with dimension ``(N, N)``
+        overwrite_a (bool): Allow overwriting data in ``a`` (may enhance
+            performance)
+        check_finite (bool): Whether to check that the input matrices contain
+            only finite numbers. Disabling may give a performance gain, but may
+            result in problems (crashes, non-termination) if the inputs do
+            contain infinities or NaNs.
+
+    Returns:
+        tuple:
+            ``(lu, piv)`` where ``lu`` is a :class:`cupy.ndarray`
+            storing ``U`` in its upper triangle, and ``L`` without
+            unit diagonal elements in its lower triangle, and `piv` is
+            a :class:`cupy.ndarray` storing pivot indices representing
+            permutation matrix ``P``.
+
+    .. seealso:: :func:`scipy.linalg.lu_factor`
+    """
+
+    if not cuda.cusolver_enabled:
+        raise RuntimeError('Current cupy only supports cusolver in CUDA 8.0')
+
+    util._assert_cupy_array(a)
```

Comment:
`scipy.linalg.lu_factor` accepts also `array_like` objects.
```suggestion
    a = cupy.asarray(a)
```